app/assets/javascripts/Components/Result/file_viewer.jsx

```diff
@@ -147,15 +147,16 @@ export class FileViewer extends React.Component {
         annotationFocus={this.props.annotationFocus}
         {...commonProps}
       />;
-    } else if (this.state.type !== '') {
+    } else if (!['binary', ''].includes(this.state.type)) {
       return <TextViewer
         type={this.state.type}
         content={this.state.content}
         focusLine={this.props.focusLine}
         {...commonProps}
       />;
-    } else {
-      return '';
-    }
+    } else if (this.state.type === 'binary'){
```

Comment:
Move this check after the previous one, that way you won't have to change any of the other code.